rep-2008.rst

```diff
@@ -0,0 +1,597 @@
+  +------------v---+ +--------v-------+         |               |
+  |                | |                |         |               |
+  | 3.2 accelerate | | 3.1 accelerate <---------> 3. hardware   |
+  |     graph      | |     nodes      |  trace  |  acceleration |
+  |                | |                |  nodes  |               <-+
+  +----------------+ +----------------+         |               | |
+                                                +---------------+ |
+                                                                  |
+                                                                  | CPU
+                                                                  | tracing
+                                                +--------------+  |
+                    +----------------+  rebuild |              |  |
+                    |                +---------->              |  |
+  start  +----------> 1. trace graph |          | 2. benchmark +--+
+                    |                |          |    CPU       |
+                    +----+------^--^-+          |              |
+                         |      |  |            +-------+------+
+                         |      |  |                    |
+                         +------+  |                    |
+                           LTTng   +--------------------+
+                                       re-instrument
+
+
+
+
+
+
+
+The following proposes a methodology to analyze a ROS 2 application and design appropriate acceleration:
+
+1. instrument both the core components of ROS 2 and the target kernels using `LTTng <https://lttng.org/>`_. Refer to `ros2_tracing <https://gitlab.com/ros-tracing/ros2_tracing>`_ for tools, documentation and ROS 2 core layers tracepoints;
+2. trace and benchmark the kernels on the CPU to establish a compute baseline;
+3. develop a hardware accelerated implementation on alternate hardware (e.g., GPU, FPGA, etc):
+
+   - **3.1** accelerate computations at the Node or Component level for each one of those identified in **2.** as good candidates.
+   - **3.2** accelerate inter-Node exchanges and reduce the overhead of the ROS 2 message-passing system across all its abstraction layers.
+
+4. trace, benchmark against the CPU baseline, and improve the accelerated implementation.
+
+The proposed ROS 2 methodology for hardware acceleration is demonstrated in [7]_ and [8]_.
+
+
+.. _acceleration examples:
+
+Acceleration examples
+^^^^^^^^^^^^^^^^^^^^^
+
+For the sake of illustrating maintainers and consumers how to build their own acceleration kernels and guarantee interoperability across technologies, a ROS 2 meta-package named `acceleration_examples <https://github.com/ros-acceleration/acceleration_examples>`_ will be maintained and made available. This meta-package will contain various packages with simple common acceleration examples. Each one of these examples should support all hardware acceleration solutions complying with this REP.
```

Comment:
I think the requirement that all examples must work on all supported solutions is too strict. 
This requirement essentially means that each hardware solution provider has to adapt all examples to target its own solution, A part from the huge overhead this will impose on eventual providers,  I am not sure this is even/easily feasible. 
Rather, It is reasonable to expect that each hardware solution has at least one example in this package.

Reply:
I'm ok relaxing the wording in here. Moreover, "*full*" compliance with `acceleration_examples` is somewhat covered at https://github.com/ros-infrastructure/rep/pull/324/files#diff-f230b6aa06d86bf594d8e431300e453ad7343e8f4b1932252b6d36c62a8b5e0aR398, which requires *official* boards to support all examples. I think it's useful for the user to know this information.